src/jrd/extds/ExtDS.cpp

```diff
@@ -2260,12 +2260,9 @@ void Statement::doSetInParams(thread_db* tdbb, unsigned int count, const MetaNam
 			paramDescs.put(*jrdVar, src);
 
 			if (src)
-			{
-				if (request->req_flags & req_null)
-					src->setNull();
-				else
-					src->clearNull();
-			}
+				src->clearNull();
+
+			//// Should not src->setNull() be called when !src?
```

Comment:
I suppose the whole if (src) {} block is not needed anymore. And below !src should be used instead of srcNull.